src/main/java/uk/ac/ebi/eva/pipeline/jobs/steps/VariantLoaderStep.java

```diff
@@ -15,69 +15,119 @@
 
     @Autowired
     private JobOptions jobOptions;
 
-    @Override
-    public RepeatStatus execute(StepContribution contribution, ChunkContext chunkContext) throws Exception {
-        ObjectMap variantOptions = jobOptions.getVariantOptions();
-        ObjectMap pipelineOptions = jobOptions.getPipelineOptions();
+    @Autowired
+    private UnwindingItemStreamReader<Variant> reader;
 
-        VariantStorageManager variantStorageManager = StorageManagerFactory.getVariantStorageManager();// TODO add mongo
+    @Autowired
+    private VariantMongoWriter variantMongoWriter;
 
-        URI outdirUri = FileUtils.getPathUri(pipelineOptions.getString(JobParametersNames.OUTPUT_DIR),true);
-        URI nextFileUri = URLHelper.createUri(pipelineOptions.getString(JobParametersNames.INPUT_VCF));
+    @Bean
+    @Qualifier("variantsLoadStep")
+    public Step variantsLoadStep() throws Exception {
+        return stepBuilderFactory.get(LOAD_VARIANTS)
+                .<Variant, Variant>chunk(jobOptions.getPipelineOptions().getInt(JobParametersNames.CONFIG_CHUNK_SIZE))
+                .reader(reader)
+                .writer(variantMongoWriter)
+                .faultTolerant().skipLimit(50).skip(FlatFileParseException.class)
+                .listener(new SkippedItemListener())
+                .build();
+    }
 
-//          URI pedigreeUri = pipelineOptions.getString(JobParametersNames.INPUT_PEDIGREE) != null ? createUri(pipelineOptions.getString(JobParametersNames.INPUT_PEDIGREE)) : null;
-        Path output = Paths.get(outdirUri.getPath());
-        Path input = Paths.get(nextFileUri.getPath());
-        Path outputVariantJsonFile = output.resolve(input.getFileName().toString() + ".variants.json" + pipelineOptions.getString("compressExtension"));
-//          outputFileJsonFile = output.resolve(input.getFileName().toString() + ".file.json" + config.compressExtension);
-        URI transformedVariantsUri = outdirUri.resolve(outputVariantJsonFile.getFileName().toString());
+    @Bean
+    @StepScope
+    public VariantMongoWriter variantMongoWriter() throws Exception {
+        MongoOperations mongoOperations = MongoDBHelper
+                .getMongoOperations(jobOptions.getDbName(), jobOptions.getMongoConnection());
 
-        logger.info("-- PreLoad variants -- {}", nextFileUri);
-        variantStorageManager.preLoad(transformedVariantsUri, outdirUri, variantOptions);
-        logger.info("-- Load variants -- {}", nextFileUri);
-        variantStorageManager.load(transformedVariantsUri, variantOptions);
-//          logger.info("-- PostLoad variants -- {}", nextFileUri);
-//          variantStorageManager.postLoad(transformedVariantsUri, outdirUri, variantOptions);
+        return new VariantMongoWriter(jobOptions.getDbCollectionsVariantsName(),
+                                      mongoOperations,
+                                      variantToMongoDbObjectConverter());
+    }
+
+    @Bean
+    @StepScope
```

Comment:
yes. there is some configuration for the mongo writer (INCLUDE_STATS, CALCULATE_STATS, INCLUDE_SAMPLES) that are fixed by a job. the job sets up a listener (VariantOptionsConfigurerListener) which configures the mongo writer through JobOptions attributes.

The listener and the mongoWriter are lazily instantiated so only one configuration is actually performed